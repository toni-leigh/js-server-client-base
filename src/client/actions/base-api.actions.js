const handleGetResponse = ({ json, responseConstant }) => ({
  type: responseConstant,
  data: json
});

export const buildGetAction = ({
  getUrl,
  responseConstant
}) => () => dispatch => {
  return fetch(getUrl)
    .then(response => response.json())
    .then(json => dispatch(handleGetResponse({ json, responseConstant })));
};