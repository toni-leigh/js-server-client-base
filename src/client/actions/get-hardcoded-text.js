const GET_HARDCODED_TEXT = 'GET_HARDCODED_TEXT';
const HARDCODED_TEXT_RECEIVED = 'HARDCODED_TEXT_RECEIVED';

const getHardcodedTextAction = () => dispatch => {
  return fetch('http://localhost:7555/hardcoded-text')
    .then(response => response.json())
    .then(json => dispatch(hardcodedTextReceived(json)));
};

const hardcodedTextReceived = hardcodedText => ({
  type: HARDCODED_TEXT_RECEIVED,
  data: { hardcodedText }
});

export {
  getHardcodedTextAction,
  GET_HARDCODED_TEXT,
  hardcodedTextReceived,
  HARDCODED_TEXT_RECEIVED
};
