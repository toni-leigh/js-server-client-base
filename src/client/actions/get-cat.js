import { buildGetAction } from './base-api.actions.js';

const CAT_RECEIVED = 'CAT_RECEIVED';

const getCat = buildGetAction({
  getUrl: 'https://api.thecatapi.com/v1/images/search?size=full',
  responseConstant: CAT_RECEIVED
});

export {
  getCat,
  CAT_RECEIVED
};
