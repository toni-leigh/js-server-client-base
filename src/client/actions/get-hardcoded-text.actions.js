import { buildGetAction } from './base-api.actions.js';

const HARDCODED_TEXT_RECEIVED = 'HARDCODED_TEXT_RECEIVED';

const getHardcodedTextAction = buildGetAction({
  getUrl: 'http://localhost:7555/hardcoded-text',
  responseConstant: HARDCODED_TEXT_RECEIVED
});

export {
  getHardcodedTextAction,
  HARDCODED_TEXT_RECEIVED
};