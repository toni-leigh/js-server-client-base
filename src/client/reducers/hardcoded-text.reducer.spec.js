import { fromJS } from 'immutable';

import hardcodedTextReducer from './hardcoded-text.reducer';
import { HARDCODED_TEXT_RECEIVED } from '../actions/get-hardcoded-text.actions';

test('hardcodedText reducer without state uses default', () => {
  expect(hardcodedTextReducer(undefined, { type: 'STUBBED' })).toMatchSnapshot();
});

test('HARDCODED_TEXT_RECEIVED', () => {
  const state = fromJS('');
  const action = { type: HARDCODED_TEXT_RECEIVED, data: 'Test Hardcoded Text' };
  expect(hardcodedTextReducer(state, action)).toMatchSnapshot();
});