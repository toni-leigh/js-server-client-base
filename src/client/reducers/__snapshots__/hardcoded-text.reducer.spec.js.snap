// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HARDCODED_TEXT_RECEIVED 1`] = `"Test Hardcoded Text"`;

exports[`hardcodedText reducer without state uses default 1`] = `""`;
